_summaries/ramachandran2017searching_2.md

```diff
@@ -1,19 +1,63 @@
 ---
 layout: summary
 title: Summary
-paper: {{paper_tag}}
+paper: ramachandran2017searching
 # Please fill out info below
-author: # Your GitHub id
-score: # How did you like this paper 0(dislike) to 10(love)
+author: liaojh1998
+score: 6
 ---
 
-TODO: Summarize the paper:
 * What is the core idea?
+
+    This paper proposed a reinforcement learning-based search method for finding good activation functions, then analyzed and evaluated the best found function, Swish, which is defined as $$x \cdot \sigma(\beta x)$$. According to evaluations, Swish is found to reliably outperform ReLU and multiple other activation functions in a wide variety of tasks and networks.
+
 * How is it realized (technically)?
+
+    * _Activation Function Search_
+
+        The authors restricted the search space to functions composed from functions that takes in 1 or 2 scalar values and output a scalar. This is done by composing from multiple repetitions of the "core unit":
+
+        <p align="center">
+            <img src="https://d3i71xaburhd42.cloudfront.net/c8c4ab59ac29973a00df4e5c8df3773a3c59995a/2-Figure1-1.png" width="30%" height="30%">
+        </p>
+
+        Note that the inputs of the unary functions are restricted to the preactivation $$x$$ and the binary function outputs. This way, the activation function can always take a scalar value as input and output a single scalar value, like ReLU.
+
+        The choice of the search algorithm depends on the size of the search space. For small search spaces, the algorithm exhaustively enumerate the entire search space, and maintain a list of top performing activation functions ordered by validation accuracy. For large search spaces, the algorithm use an RNN controller to "predict" components of the activation function:
+
+        <p align="center">
+            <img src="https://d3i71xaburhd42.cloudfront.net/c8c4ab59ac29973a00df4e5c8df3773a3c59995a/3-Figure2-1.png" width="30%" height="30%">
+        </p>
+
+        The RNN predicts a single component of the activation function at each timestep, and each prediction is fed back as input to "predict" the component of the next timestep in an autoregressive fashion. This network is trained with reinforcement learning (Proximal Policy Optimization, [Schulman et al., 2017](https://arxiv.org/abs/1707.06347)) to output functions that have high validation accuracies by using validation accuracy as the reward. To reduce search time, the search algorithm batched candidate functions and aggregated their validation accuracies to update the search algorithm.
+
+    * _Search Evaluations_
+        
+        The searches are conducted by training ResNet-20 ([He et al., 2016a](https://ieeexplore.ieee.org/document/7780459)) on the CIFAR-10 ([Krizhevsky & Hinton, 2009](https://www.cs.toronto.edu/~kriz/learning-features-2009-TR.pdf)) by substituting each activation function instead of ReLU. Every network is trained for 10K steps.
```

Comment:
Consider mentioning the use of "child networks" more explicitly, as well as the use of distributed training to parallelize computation.

Reply:
Done!